src/ILLink.RoslynAnalyzer/RequiresUnreferencedCodeAnalyzer.cs

```diff
@@ -84,15 +85,22 @@ protected override ImmutableArray<ISymbol> GetSpecialIncompatibleMembers (Compil
 				incompatibleMembers.AddRange (methodInfoType.GetMembers ("MakeGenericMethod").OfType<IMethodSymbol> ());
 			}
 
+			var expressionCallType = compilation.GetTypeByMetadataName ("System.Linq.Expressions.Expression");
+			if (expressionCallType != null) {
+				incompatibleMembers.AddRange (expressionCallType.GetMembers ("Call").OfType<IMethodSymbol> ()
+					.Where (m => m.Parameters.Length == 4 && m.Parameters[0] is IParameterSymbol parameterSymbol
```

Comment:
The list of incompatible member purposes is to say "RequiresUnreferencedCodeAnalyzer please don't analyze this call" then the dataflow will handle the intrinsic and decide if it needs to warn or not. You are restricting the number of parameters and type of the parameter, which means that RequiresUnreferencedCodeAnalyzer will only bypass that case and for everything else will warn. Im afraid that dataflow will do something similar and we end up generating two warnings, I think what we want is:
```diff
- incompatibleMembers.AddRange (expressionCallType.GetMembers ("Call").OfType<IMethodSymbol> ()
-					.Where (m => m.Parameters.Length == 4 && m.Parameters[0] is IParameterSymbol parameterSymbol
-					&& parameterSymbol.Type?.ContainingNamespace?.Name == "System" && parameterSymbol.Type?.Name == "Type"));
+ incompatibleMembers.AddRange (expressionCallType.GetMembers ("Call").OfType<IMethodSymbol> ());
```

Reply:
unfortunately, there are lots of these Call methods and only 1 should be handled intrinsically. The code mimics the [linker ](https://github.com/dotnet/linker/blob/feature/damAnalyzer/src/ILLink.Shared/TrimAnalysis/Intrinsics.cs#L59-L63)version. There are some other expression members that need special handling and I'm taking a look at those as part of #2578

Reply:
I think https://github.com/dotnet/linker/issues/2578 will require a more general solution but I am fine with this change doing the minimum necessary to get the suppression tests working.